qiskit/primitives/fidelity/base_fidelity.py

```diff
@@ -0,0 +1,84 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Base fidelity primitive
+"""
+
+from abc import ABC, abstractmethod
+from typing import List, Union, Optional
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import ParameterVector
+
+
+class BaseFidelity(ABC):
+    """
+    Implements the interface to calculate fidelities.
+    """
+
+    def __init__(
+        self,
+        left_circuit: QuantumCircuit,
+        right_circuit: QuantumCircuit,
+    ) -> None:
+        r"""
+        Initializes the class to evaluate the fidelities defined as
+            :math:`|\langle\psi(x)|\phi(y)\rangle|^2`,
+        where x and y are parametrizations of the circuits :math:`\psi` and :math:`\phi`.
+        Args:
+            left_circuit: (Parametrized) quantum circuit :math:`|\psi\rangle`
+            right_circuit: (Parametrized) quantum circuit :math:`|\phi\rangle`
+        Raises:
+            ValueError: left_circuit and right_circuit don't have the same number of qubits
```

Comment:
I think we generally associate circuits with unitaries preparing states, not as states themselves, as we don't specify the initial state of the qubits 🙂 
```suggestion
        r"""Initializes the class to evaluate the fidelities defined as
        
            :math:`|\langle\psi(x)|\phi(y)\rangle|^2`,
            
        where :math:`x` and :math:`y` are optional parametrizations of the
        states :math:`\psi` and :math:`\phi` prepared by the circuits
        ``left_circuit`` and ``right_circuit``, respectively.
        
        Args:
            left_circuit: (Parametrized) quantum circuit preparing :math:`|\psi\rangle`.
            right_circuit: (Parametrized) quantum circuit preparing :math:`|\phi\rangle`.
            
        Raises:
            ValueError: ``left_circuit`` and ``right_circuit`` don't have the same number of qubits.
```